### DISCUSSION TITLE
API Models expected

### DISCUSSION THREAD

**Discussion by deep-mm** [CSC510-Group-5/URL-Shortner#19] [General]
@akk5597 @psurana-ncsu 
What is the URL model that Post API expects from the front end?

FYI @alunavat

**Comment by akk5597** [comment]
For `/new`, the POST request should look like this
```json
{
    "long_url": <long URL entered by user>
}
```

And the response would be 
```json
{
    "long_url": <long URL entered by user>, 
    "special_code": <special code generated by system for verification during update or remove>, 
    "stub": <stub that would be used for short link>
}
```

For `/stub/{stub}`, there would only be a GET request that would return a redirection.

If need be, we can change it to be POST on `/stub/` that needs
```json
{
    "stub": <stub>
}
```
And the response would be 
```json
{
    "long_url": <long URL entered by user>
}
```
Let me know which one is more convenient @deep-mm, cc @alunavat

**Reply by alunavat** [reply]
@akk5597 -> Weren't we also supposed to have an expiration date?

**Reply by deep-mm** [reply]
@akk5597 leta directly redirect users to long url, that should serve the purpose.

@alunavat lets keep expiration date at low priority for now and add it later if time remains.

**Reply by alunavat** [reply]
Sure, that works @deep-mm
